lib/gridstore/src/blob.rs

```diff
@@ -3,3 +3,57 @@ pub trait Blob {
 
     fn from_bytes(bytes: &[u8]) -> Self;
 }
+
+impl Blob for i64 {
+    fn to_bytes(&self) -> Vec<u8> {
+        self.to_le_bytes().to_vec()
+    }
+
+    fn from_bytes(data: &[u8]) -> Self {
+        i64::from_le_bytes(
+            data.try_into()
+                .expect("Slice with incorrect length for i64"),
+        )
+    }
+}
+
+impl Blob for u128 {
+    fn to_bytes(&self) -> Vec<u8> {
+        self.to_le_bytes().to_vec()
+    }
+
+    fn from_bytes(data: &[u8]) -> Self {
+        u128::from_le_bytes(
+            data.try_into()
+                .expect("Slice with incorrect length for u128"),
+        )
+    }
+}
+
+impl Blob for f64 {
+    fn to_bytes(&self) -> Vec<u8> {
+        self.to_le_bytes().to_vec()
+    }
+
+    fn from_bytes(data: &[u8]) -> Self {
+        f64::from_le_bytes(
+            data.try_into()
+                .expect("Slice with incorrect length for f64"),
+        )
+    }
+}
```

Comment:
Looks like implementation would have less repetition with a macro
```suggestion
macro_rules! impl_blob_le_bytes {
    ($type:ty) => {
        impl Blob for $type {
            fn to_bytes(&self) -> Vec<u8> {
                self.to_le_bytes().to_vec()
            }

            fn from_bytes(data: &[u8]) -> Self {
                <$type>::from_le_bytes(
                    data.try_into()
                        .expect(&format!("Slice with incorrect length for {}", stringify!($type))),
                )
            }
        }
    }
}

impl_blob_le_bytes!(i64);
impl_blob_le_bytes!(f64);
impl_blob_le_bytes!(u128);
```

Reply:
Although, actually, if it is always going to be used within a `Vec<T>`, the macro can already be augmented by implementing for the vec directly. This would avoid intermediate `to_vec()` calls